Windows support

hishamhm opened the pull request:
Installs from the LuaRocks source package on Unix and uses the precompiled binaries on Windows.

CI: Test matrix runs it with Lua 5.1 (oldest supported), 5.4 (newest supported) and LuaJIT; being based on gh-actions-lua, it uses MSVC for PUC-Rio Lua runs and Mingw for the LuaJIT run. In principle _this_ action should work with any combination of Lua version and supported Windows compiler, even though gh-actions-lua currently doesn't (it would be rather annoying to support the full Windows matrix on gh-actions-lua because the LuaJIT Makefile works out of the box with Mingw and my handwritten implementation for building PUC Lua targets MSVC and would have to be rewritten for Mingw).

Closes #13.

hishamhm committed:
Windows support

hishamhm committed:
Display full configuration in CI run

`luarocks path` on its own does not set the paths, it just
displays the variables to be set. `luarocks config` is a more
useful test as it prints the whole internal configuration.

msteinbeck commented:
Any progress here?

hishamhm commented:
@leafo You can see it running on my fork's CI here: https://github.com/hishamhm/gh-actions-luarocks/actions/runs/2758104388

msteinbeck commented:
So this PR could be merged?

magneto538 commented:
Up — can this be merged? Seems like it should address https://github.com/leafo/gh-actions-luarocks/issues/17 and I'd need that.

dargueta commented:
This breaks on Cygwin, unfortunately, since it's still looking for `cl` instead of `gcc`.

magneto538 commented:
Up?

TheComamba commented:
@hishamhm Could you make a release on your fork? Then we could simply use your version for the moment.

mrcjkb commented on main.js:
```diff
@@ -50,52 +74,83 @@ async function main() {
 
-  if (luaPath != "") {
-    core.exportVariable("LUA_PATH", ";;" + luaPath)
+  if (lrPath != "") {
+    core.exportVariable("LUA_PATH", ";;" + lrPath)
   }
 
-  if (luaCpath != "") {
-    core.exportVariable("LUA_CPATH", ";;" + luaCpath)
+  if (lrCpath != "") {
+    core.exportVariable("LUA_CPATH", ";;" + lrCpath)
   }
```
I've been using this workflow, and this seems to generate a unix style path (with semicolons as separators, instead of colons).

```
> printenv
LUA_PATH=;;C:\Users\runneradmin\AppData\Roaming/luarocks/share/lua/5.1/?.lua;C:\Users\runneradmin\AppData\Roaming/luarocks/share/lua/5.1/?/init.lua

PATH=/c/Program Files/PowerShell/7:/c/Users/runneradmin/AppData/Roaming/luarocks/bin:/d/a/rocks-binaries/rocks-binaries/.luarocks/bin:/d/a/rocks-binaries/rocks-binaries/.lua/bin:/c/Program Files (x86)/Microsoft Visual Studio/2019/Enterprise/Common7/IDE/Extensions/Microsoft/IntelliCode/CLI:/c/Program Files (x86)/Microsoft Visual Studio/2019/Enterprise/VC/Tools/MSVC/14.29.30133/bin/HostX64/x64:/c/Program Files (x86)/Microsoft Visual Studio/2019/Enterprise/Common7/IDE/VC/VCPackages:/c/Program Files (x86)/Microsoft Visual 
...
```

As a result, build-time dependencies can't be found (e.g. when installing `toml-edit`, which depends on `luarocks-build-rust-mlua`.

linrongbin16 replied:
@mrcjkb  did you solve this issue?

mrcjkb replied:
No, not yet.

hishamhm replied:
@mrcjkb LUA_PATH uses `;` on both Windows and Unix, and PATH uses `:` on Unix and `;` on Windows. I don't think the semicolons are the cause of your issue, but [there were issues related to build dependency lookup](https://github.com/luarocks/luarocks/commit/ea43975308abc8b5281ac8934657aa4d59bcdd75) that were fixed in 3.10.0.

mrcjkb replied:
🙏 I can confirm it works when I add

```yaml
with:
    luarocksVersion: "3.10.0"
```

mrcjkb replied:
Actually, no it doesn't. I was looking at the Linux builder.
The Windows builder still fails when trying to use the `luarocks-build-rust-mlua` build backend: 

```
Error: Failed initializing build back-end for build type 'rust-mlua': module 'luarocks.build.rust-mlua' not found:
	No LuaRocks module found for luarocks.build.rust-mlua
```

https://github.com/nvim-neorocks/rocks-binaries/actions/runs/8168782585/job/22331528480#step:11:1256

hishamhm replied:
@mrcjkb Please try again bumping the version number to `3.11.0`! I've uploaded a pre-release build but I haven't announced it yet. :shushing_face: :wink:

mrcjkb replied:
That has fixed it! :pray: :pray: :pray: 
:tada: :rocket:

magneto538 commented:
@hishamhm I've been trying to use your fork on `windows-latest`, but I'm still getting 

`Error: Failed to install LuaRocks: Error: Unable to locate executable file: lua. Please verify either the file path exists or the file can be found within a directory specified by the PATH environment variable. Also verify the file has a valid extension for an executable file.`

Do you have any advice as to how I might fix this?

hishamhm committed:
deps: add semver, update @actions dependencies

Just added "semver" and ran `make vendor`.

hishamhm committed:
Windows: version-gate mingw workaround for LuaRocks < 3.9.2

williamboman committed:
fix: fix ReferenceError on windows

magneto538 commented:
@hishamhm , trying your fork off the tip of `master`. Upon attempting to install packages via Luarocks on `windows-latest` using the default Win shell, I'm getting `luarocks: command not found`. Any advice?

hishamhm commented:
@magneto538 no idea, sorry! it seems to be working for others, so I don't even know what to suggest...

jonasgeiler commented:
Great work on the windows support and thank you for your fork @hishamhm - everything seems to work perfectly except when I try to require `luarocks.loader`, which can't be found unfortunately... It works on Linux.

Here is my workflow:
```yaml
on:
  workflow_dispatch:

jobs:
  test-luarocks-loader:
    name: Test LuaRocks Loader
    runs-on: windows-latest

    steps:
      - name: Setup Microsoft Visual C++ Developer Command Prompt
        uses: ilammy/msvc-dev-cmd@v1.13.0

      - name: Setup Lua
        uses: leafo/gh-actions-lua@v10.0.0

      - name: Setup LuaRocks
        uses: hishamhm/gh-actions-luarocks@master

      - name: Test require luarocks loader
        run: lua -e "require('luarocks.loader');"
```
And here is the output of the last step above:
```
2024-04-06T06:47:45.9623061Z D:/a/lua-fenster/lua-fenster/.lua/bin\lua.exe: (command line):1: module 'luarocks.loader' not found:
2024-04-06T06:47:45.9624644Z 	no field package.preload['luarocks.loader']
2024-04-06T06:47:45.9626041Z 	no file 'D:\a\lua-fenster\lua-fenster\.lua\bin\lua\luarocks\loader.lua'
2024-04-06T06:47:45.9627431Z 	no file 'D:\a\lua-fenster\lua-fenster\.lua\bin\lua\luarocks\loader\init.lua'
2024-04-06T06:47:45.9629064Z 	no file 'D:\a\lua-fenster\lua-fenster\.lua\bin\luarocks\loader.lua'
2024-04-06T06:47:45.9631579Z 	no file 'D:\a\lua-fenster\lua-fenster\.lua\bin\luarocks\loader\init.lua'
2024-04-06T06:47:45.9633156Z 	no file 'D:\a\lua-fenster\lua-fenster\.lua\bin\..\share\lua\5.4\luarocks\loader.lua'
2024-04-06T06:47:45.9634704Z 	no file 'D:\a\lua-fenster\lua-fenster\.lua\bin\..\share\lua\5.4\luarocks\loader\init.lua'
2024-04-06T06:47:45.9636184Z 	no file '.\luarocks\loader.lua'
2024-04-06T06:47:45.9637341Z 	no file '.\luarocks\loader\init.lua'
2024-04-06T06:47:45.9638632Z 	no file 'C:\Users\runneradmin\AppData\Roaming/luarocks/share/lua/5.4/luarocks\loader.lua'
2024-04-06T06:47:45.9640845Z 	no file 'C:\Users\runneradmin\AppData\Roaming/luarocks/share/lua/5.4/luarocks\loader/init.lua'
2024-04-06T06:47:45.9642556Z 	no file 'D:\a\lua-fenster\lua-fenster\.lua\bin\luarocks\loader.dll'
2024-04-06T06:47:45.9644015Z 	no file 'D:\a\lua-fenster\lua-fenster\.lua\bin\..\lib\lua\5.4\luarocks\loader.dll'
2024-04-06T06:47:45.9645342Z 	no file 'D:\a\lua-fenster\lua-fenster\.lua\bin\loadall.dll'
2024-04-06T06:47:45.9646255Z 	no file '.\luarocks\loader.dll'
2024-04-06T06:47:45.9647493Z 	no file 'C:\Users\runneradmin\AppData\Roaming/luarocks/lib/lua/5.4/luarocks\loader.dll'
2024-04-06T06:47:45.9648931Z 	no file 'D:\a\lua-fenster\lua-fenster\.lua\bin\luarocks.dll'
2024-04-06T06:47:45.9650260Z 	no file 'D:\a\lua-fenster\lua-fenster\.lua\bin\..\lib\lua\5.4\luarocks.dll'
2024-04-06T06:47:45.9651596Z 	no file 'D:\a\lua-fenster\lua-fenster\.lua\bin\loadall.dll'
2024-04-06T06:47:45.9652592Z 	no file '.\luarocks.dll'
2024-04-06T06:47:45.9657461Z 	no file 'C:\Users\runneradmin\AppData\Roaming/luarocks/lib/lua/5.4/luarocks.dll'
2024-04-06T06:47:45.9682363Z stack traceback:
2024-04-06T06:47:45.9719912Z 	[C]: in function 'require'
2024-04-06T06:47:45.9743053Z 	(command line):1: in main chunk
2024-04-06T06:47:45.9771143Z 	[C]: in ?
```

I'm not really using the luarocks loader so it's not a critical issue, just noticed it and wanted to let you know.

magneto538 commented:
I have a problem too. My workflow requires the `lfs` Lua module. When attempting to run a script that requires the `lfs` module after having installed it, I'm getting `lfs module not found`, as per log attached. Any advice?

```
name: foo

on:
  workflow_dispatch:

jobs:
  tests:
    runs-on: windows-latest
    steps:
      - uses: actions/checkout@v4.1.1

      - name: Setup Microsoft Visual C++ Developer Command Prompt
        uses: ilammy/msvc-dev-cmd@v1.13.0

      - name: Setup Lua
        uses: leafo/gh-actions-lua@v10.0.0

      - name: Setup LuaRocks
        uses: hishamhm/gh-actions-luarocks@master
        with:
          luarocksVersion: 3.11.0

      - name: Environment setup
        run: |
          luarocks install luacheck
          luacheck --version
          luarocks install busted
          luarocks install luafilesystem
          luarocks install serpent  
          lua runScript.lua ## this script requires luafilesystem (lfs)
```

Log from Environment setup step:

```
Run luarocks install luacheck
  
lfs.c
Microsoft (R) Incremental Linker Version 14.38.33135.0
Copyright (C) Microsoft Corporation.  All rights reserved.
   Creating library C:\Users\RUNNER~1\AppData\Local\Temp\luarocks_build-LuaFileSystem-1.8.0-1-8365456\lfs.lib and object C:\Users\RUNNER~1\AppData\Local\Temp\luarocks_build-LuaFileSystem-1.8.0-1-8365456\lfs.exp
No existing manifest. Attempting to rebuild...
Installing https://luarocks.org/luacheck-1.1.2-1.src.rock
Missing dependencies for luacheck 1.1.2-1:
   argparse >= 0.6.0 (not installed)
   luafilesystem >= 1.6.3 (not installed)
luacheck 1.1.2-1 depends on lua >= 5.1 (5.4-1 provided by VM: success)
luacheck 1.1.2-1 depends on argparse >= 0.6.0 (not installed)
Installing https://luarocks.org/argparse-0.7.1-1.src.rock
argparse 0.7.1-1 depends on lua >= 5.1, < 5.5 (5.4-1 provided by VM: success)
argparse 0.7.1-1 is now installed in C:\Users\runneradmin\AppData\Roaming\luarocks (license: MIT)
luacheck 1.1.2-1 depends on luafilesystem >= 1.6.3 (not installed)
Installing https://luarocks.org/luafilesystem-1.8.0-1.src.rock
luafilesystem 1.8.0-1 depends on lua >= 5.1 (5.4-1 provided by VM: success)
cl /nologo /MD /O2 -c -Fosrc/lfs.obj -ID:/a/cradle-products/cradle-products/.lua/include src/lfs.c
link -dll -def:lfs.def -out:C:\Users\RUNNER~1\AppData\Local\Temp\luarocks_build-LuaFileSystem-1.8.0-1-8365456\lfs.dll D:\a\cradle-products\cradle-products\.lua\lib\lua54.lib src/lfs.obj
luafilesystem 1.8.0-1 is now installed in C:\Users\runneradmin\AppData\Roaming\luarocks (license: MIT/X11)
luacheck 1.1.2-1 is now installed in C:\Users\runneradmin\AppData\Roaming\luarocks (license: MIT)
D:\a\cradle-products\cradle-products\.lua\bin\lua.exe: error loading module 'lfs' from file 'C:\Users\runneradmin\AppData\Roaming\luarocks\lib\lua\5.4\lfs.dll':
	The specified module could not be found.
stack traceback:
	[C]: in ?
	[C]: in function 'require'
	...n\AppData\Roaming\luarocks\share\lua\5.4\luacheck\fs.lua:3: in main chunk
	[C]: in function 'require'
	...ppData\Roaming\luarocks\share\lua\5.4\luacheck\cache.lua:1: in main chunk
	[C]: in function 'require'
	...AppData\Roaming\luarocks\share\lua\5.4\luacheck\main.lua:2: in main chunk
	[C]: in function 'require'
	...cks\lib\luarocks\rocks-5.4\luacheck\1.1.2-1\bin\luacheck:2: in main chunk
	[C]: in ?
core.c
compat.c
time.c
environment.c
random.c
link -dll -def:core.def -out:C:\Users\RUNNER~1\AppData\Local\Temp\luarocks_build-lua-term-0.8-1-[62](https://github.com/cradleapps/cradle-products/actions/runs/8586910354/job/23529915182?pr=1777#step:8:63)19236\term\core.dll D:\a\cradle-products\cradle-products\.lua\lib\lua54.lib core.obj
lua-term 0.8-1 is now installed in C:\Users\runneradmin\AppData\Roaming\luarocks (license: MIT/X11)
busted 2.2.0-1 depends on penlight >= 1.3.2 (not installed)
Installing https://luarocks.org/penlight-1.13.1-1.src.rock
penlight 1.13.1-1 depends on luafilesystem (1.8.0-1 installed: success)
penlight 1.13.1-1 is now installed in C:\Users\runneradmin\AppData\Roaming\luarocks (license: MIT/X11)
busted 2.2.0-1 depends on mediator_lua >= 1.1.1 (not installed)
Installing https://luarocks.org/mediator_lua-1.1.2-0.rockspec
mediator_lua 1.1.2-0 depends on lua >= 5.1 (5.4-1 provided by VM: success)
mediator_lua 1.1.2-0 is now installed in C:\Users\runneradmin\AppData\Roaming\luarocks (license: MIT <http://opensource.org/licenses/MIT>)
busted 2.2.0-1 is now installed in C:\Users\runneradmin\AppData\Roaming\luarocks (license: MIT <http://opensource.org/licenses/MIT>)
Installing https://luarocks.org/luafilesystem-1.8.0-1.src.rock
luafilesystem 1.8.0-1 is already installed in C:\Users\runneradmin\AppData\Roaming\luarocks
Use --force to reinstall.
Cloning into 'serpent'...
Note: switching to 'd78[68](https://github.com/cradleapps/cradle-products/actions/runs/8586910354/job/23529915182?pr=1777#step:8:69)3597606c6e13a1fed039bc91d86eb8f600f'.
You are in 'detached HEAD' state. You can look around, make experimental
changes and commit them, and you can discard any commits you make in this
state without impacting any branches by switching back to a branch.
If you want to create a new branch to retain commits you create, you may
do so (now or later) by using -c with the switch command. Example:
  git switch -c <new-branch-name>
Or undo this operation with:
  git switch -
Turn off this advice by setting config variable advice.detachedHead to false
Installing https://luarocks.org/serpent-0.30-2.rockspec
D:/a/cradle-products/cradle-products/.lua/bin\lua.exe: error loading module 'lfs' from file 'C:\Users\runneradmin\AppData\Roaming\luarocks\lib\lua\5.4\lfs.dll':
	The specified module could not be found.
stack traceback:
	[C]: in ?
	[C]: in function 'require'
	ActionsHelpers/runCradleEngineInstaller.lua:8: in main chunk
	[C]: in ?
serpent 0.30-2 depends on lua >= 5.1, < 5.5 (5.4-1 provided by VM: success)
serpent 0.30-2 is now installed in C:\Users\runneradmin\AppData\Roaming\luarocks (license: MIT)
```

Tieske commented:
@magneto538 are you using @hishamhm s fork? That should fix that issue (see https://github.com/lunarmodules/luasystem/pull/17#issuecomment-1979530101) unless you're hitting a new issue.

magneto538 commented:
@Tieske Yes, I am, as per the code snippet I've posted.

magneto538 commented:
@hishamhm bump in case you missed it ^

hishamhm commented:
@magneto538 as a workaround, try adding `luarocks config fs_use_modules false` before `luarocks install luafilesystem`

magneto538 commented:
@hishamhm thanks. Now `lfs` would install correctly as per Luarocks' output, however it still fails when called from a script executed after setting up the environment (see [the first snippet in this comment](https://github.com/leafo/gh-actions-luarocks/pull/14#issuecomment-2041344643) for reference):

```
D:/a/myrepo/myrepo/.lua/bin\lua.exe: runScript.lua:8: module 'lfs' not found:
	no field package.preload['lfs']
	no file 'D:\a\myrepo\myrepo\.lua\bin\lua\lfs.lua'
	no file 'D:\a\myrepo\myrepo\.lua\bin\lua\lfs\init.lua'
	no file 'D:\a\myrepo\myrepo\.lua\bin\lfs.lua'
	no file 'D:\a\myrepo\myrepo\.lua\bin\lfs\init.lua'
	no file 'D:\a\myrepo\myrepo\.lua\bin\..\share\lua\5.4\lfs.lua'
	no file 'D:\a\myrepo\myrepo\.lua\bin\..\share\lua\5.4\lfs\init.lua'
	no file '.\lfs.lua'
	no file '.\lfs\init.lua'
	no file 'C:\Users\runneradmin\AppData\Roaming\luarocks\share\lua\5.4\lfs.lua'
	no file 'C:\Users\runneradmin\AppData\Roaming\luarocks\share\lua\5.4\lfs\init.lua'
	no file 'D:\a\myrepo\myrepo\.lua\bin\lfs.dll'
	no file 'D:\a\myrepo\myrepo\.lua\bin\..\lib\lua\5.4\lfs.dll'
	no file 'D:\a\myrepo\myrepo\.lua\bin\loadall.dll'
	no file '.\lfs.dll'
	no file 'C:\Users\runneradmin\AppData\Roaming\luarocks\lib\lua\5.4\lfs.dll'
stack traceback:
	[C]: in function 'require'
	runScript.lua:8: in main chunk
	[C]: in ?
```

hishamhm commented:
@magneto538 `eval $(luarocks path)` ?

magneto538 commented:
@hishamhm no luck with that one:

```
eval: D:\a\_temp\019bc3a1-dc0e-4b98-879f-3321a897784e.ps1:8
Line |
   8 |  eval $(luarocks path)
     |  ~~~~
     | The term 'eval' is not recognized as a name of a cmdlet, function, script file, or executable program. Check the
     | spelling of the name, or if a path was included, verify that the path is correct and try again.
```

jonasgeiler commented:
@magneto538 he mistakenly gave you the code for Unix, here's how you do it on Windows CMD:
```cmd
luarocks path > "%temp%\_lrp.bat"
call "%temp%\_lrp.bat" && del "%temp%\_lrp.bat"
```
You might have to set `shell: cmd` in your GitHub Actions job to use CMD instead of the default PowerShell.

(You can find this command by running `luarocks path --help`)

magneto538 commented:
@jonasgeiler thanks! Actually I did try switching to `bash` before posting (while keeping the Unix `eval` call provided by @hishamhm) and got the exact same output I'm getting with PowerShell using your Windows command instead:

```
Error: Failed installing dependency: https://luarocks.org/luafilesystem-1.8.0-1.src.rock - Build error: Failed compiling module lfs.dll
   argparse >= 0.6.0 (not installed)
   luafilesystem >= 1.6.3 (not installed)
```

So nope, not working for me either.

Tieske commented:
@magneto538 GHA defaults to PowerShell as the shell, but you can set it up to switch to `cmd` iirc. Try adding `shell: cmd`.

@jonasgeiler I originally also thought the dll was wrong, hence we added this: https://github.com/lunarmodules/luasystem/pull/17/files#diff-5c3fa597431eda03ac3339ae6bf7f05e1a50d6fc7333679ec38e21b337cb6721R66 
turned out dll was fine in my case. Maybe the code can help you debug your problem.

jonasgeiler commented:
> @Tieske wrote:
> (at)jonasgeiler I originally also thought the dll was wrong, hence we added this: https://github.com/lunarmodules/luasystem/pull/17/files#diff-5c3fa597431eda03ac3339ae6bf7f05e1a50d6fc7333679ec38e21b337cb6721R66 turned out dll was fine in my case. Maybe the code can help you debug your problem.

Thank you for showing me the workflow you use for luasystem! I tried adapting some of your steps and I got the whole thing working once I switched to `hishamhm/gh-actions-lua@master` instead of `leafo/gh-actions-lua@v10.0.0` 🤔 Seems there are some flaws with leafo's Lua action aswell...

Here is the commit that finally fixed my problem:
https://github.com/jonasgeiler/lua-fenster/commit/5f41f11dd87cf7843117a7d8a786d1fb17baa487

Unfortunately I didn't get LuaJIT tests to work on Windows but as long as all the other versions work I think it's fine for now.

Tieske commented:
> Unfortunately I didn't get LuaJIT tests to work on Windows but as long as all the other versions work I think it's fine for now.

The trick in `luasystem` is that PuC Lua and LuaJIT are build using different toolchains. One uses MinGW, the other MSVC. So might be worthwhile to try and get it to work, as it provides better test coverage.

jonasgeiler commented:
> @Tieske wrote: The trick in `luasystem` is that PuC Lua and LuaJIT are build using different toolchains. One uses MinGW, the other MSVC. So might be worthwhile to try and get it to work, as it provides better test coverage.

Yeah I did try to skip the MSVC step and force MingW with `shell: bash` and everything, just like `luasystem` does, but to no success... I might have missed something though and will probably revisit this problem in the future when my library gains more popularity. I could confirm that it works on LuaJIT on Windows by building it manually, so it's not a big problem.
Thanks anyways!

hishamhm committed:
force LUA_LIBDIR

hishamhm committed:
update default version

hishamhm committed:
CI: update LuaJIT rolling release using hishamhm/gh-actions-lua

hishamhm committed:
CI: use gh-actions-luarocks being tested

hishamhm committed:
docs: fix typo in luaRocksVersion attribute name

jonasgeiler commented:
@hishamhm It seems like you've commited the `node_modules` folder here... The `node_modules` folder should definitely stay out of VCS.

hishamhm committed:
[fork] update URLs

luau-project committed:
fix: set MSVCRT config for toolchains other than MSVC

bzp2010 committed:
feat: allow use git commit id as version

hishamhm committed:
bump LuaRocks version

luau-project committed:
luarocks: update to 3.12.1

luau-project committed:
LuaRocks: update to 3.12.2

luau-project committed:
ci: use repos from luarocks

luau-project committed:
ci: ignore changes on markdown files

luau-project committed:
ci: add version 5.4.8 to the matrix

luau-project committed:
docs: update versions of actions

luau-project committed:
Merge pull request #8 from luau-project/prepare-v6

LuaRocks: minor changes preparing a new v6 tag

magneto538 commented:
Anyone still waiting on this —, I was able to successfully use `gh-actions-luarocks` and `gh-actions-lua` on a Windows agent by switching over to https://github.com/luarocks/gh-actions-lua and https://github.com/luarocks/gh-actions-luarocks . @v11 and @v6 specifically work for me.

hishamhm committed:
LuaRocks: update to 3.13.0 (#10)

* LuaRocks: update to 3.13.0

* add Lua 5.5 to the matrix

* use gh-actions-luarocks from checkout on tests

---------

Co-authored-by: github-actions[bot] <41898282+github-actions[bot]@users.noreply.github.com>
Co-authored-by: luau-project <luau.project@gmail.com>

Copilot committed:
Update node16 to node20, migrate to ESM, upgrade @actions/* packages to latest major versions

Agent-Logs-Url: https://github.com/luarocks/gh-actions-luarocks/sessions/c574036f-c5df-43bb-ba90-9c73adbcd2d1

Co-authored-by: Tieske <1077043+Tieske@users.noreply.github.com>

Copilot committed:
Update runtime from node20 to node24

Agent-Logs-Url: https://github.com/luarocks/gh-actions-luarocks/sessions/fb71c753-5169-4aa7-bf1e-48bce0fd65ef

Co-authored-by: Tieske <1077043+Tieske@users.noreply.github.com>

luau-project committed:
ci: update action to setup MSVC without warnings

* At the moment, the current action to setup
  MSVC ( https://github.com/ilammy/msvc-dev-cmd )
  emits warnings due the use of EOL nodejs
  versions;
* The action's maintainer does not respond PRs;
* Apparently, there is no movement to fix such
  warnings;
* Follow upstream (leafo's action) and use
  ( https://github.com/step-security/msvc-dev-cmd ).

luau-project committed:
ci: add Lua 5.5.0 to matrix

luau-project committed:
docs: update default version to 3.13.0

luau-project committed:
docs: bump gh-actions-lua and gh-actions-luarocks

luau-project committed:
ci: move to actions `main` branch

* actions `master` branch is an ancient branch
  out of updates for 6 years;
* actions `main` branch is the default branch.
